### DISCUSSION TITLE
[IgxGrid] - How do you groupby columns that are autogenerated?

### DISCUSSION THREAD

**Discussion by mddifilippo89** [IgniteUI/igniteui-angular#12717] [Q&A] [answered]
With autogenerated columns how do you groupby columns through the UI when showGroupArea is true?

**Comment by georgianastasov** [comment, marked as answer]
Hello @mddifilippo89 

When we have auto-generated columns and we want to group them, what we can do is to handle the **ngAfterViewInit** hook and, when loading and rendering the grid, access its [columns](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#columns). We can then iterate through them and set each [groupable ](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxColumnComponent.html#groupable)property to true.
```
<igx-grid
    #grid
    [data]="data"
    [autoGenerate]="true"
    width="100%"
    height="600px"
    [showGroupArea]="true"
  ></igx-grid>
```
```
public ngAfterViewInit() {
    this.grid.columns.forEach((column) => {
         column.groupable = true;
    });
} 
```
[Here](https://stackblitz.com/edit/angular-6gdy6f?file=src/app/grid/grid-column-hiding-toolbar-sample/grid-column-hiding-toolbar-sample.component.html) could be found a small sample demonstrating the above mentioned approach.
